### PR TITLE
[chore](sql) Forbid show hidden columns and create table with hidden column

### PR DESCRIPTION
**Pull request by 924060929**
## Proposed changes

Forbid show hidden columns and create table with hidden column

### CODE REVIEW

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateTableStmt.java [hunk 1]
```diff
@@ -319,6 +319,11 @@ public void analyze(Analyzer analyzer) throws UserException {
             if (Objects.equals(columnDef.getType(), Type.ALL)) {
                 throw new AnalysisException("Disable to create table with `ALL` type columns.");
             }
+            String columnNameUpperCase = columnDef.getName().toUpperCase();
+            if (columnNameUpperCase.startsWith("__DORIS_") && columnNameUpperCase.endsWith("_SIGN__")) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/CreateTableStmt.java:323]: should use invisable column list in `org.apache.doris.catalog.Column`

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by feiniaofeiafei — APPROVED** [review]

**Comment by 924060929** [comment]
run buildall

**Review by xzj7019 — APPROVED** [review]

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**